tmt/result.py

```diff
@@ -376,14 +372,8 @@ def from_test_invocation(
 
         # Saving identifiable information for each test case so we can match them
         # to Polarion/Nitrate/other cases and report run results there
-        # TODO: would an exception be better? Can test.id be None?
         ids = ids or {}
-        default_ids: ResultIds = {tmt.identifier.ID_KEY: invocation.test.id}
-
-        for key in EXTRA_RESULT_IDENTIFICATION_KEYS:
-            value: Any = cast(Any, invocation.test.node.get(key))
-
-            default_ids[key] = None if value is None else str(value)
+        default_ids: ResultIds = invocation.test.get_ids()
 
         default_ids.update(ids)
         ids = default_ids
```

Comment:
We can do something nice here:

```python
ids = {
  **invocation.test.ids,
  **(ids or {})
}
```

The update of `default_ids` and then `ids = default_ids` is really scary, and since you are in the area...